### DISCUSSION TITLE
2023 Elyra Development Roadmap

### DISCUSSION THREAD

**Discussion by ptitzler** [elyra-ai/elyra#3016] [General]
- 3.14.x patch releases as necessary

## Q 1 2023
 - Begin work on 4.0.0 
    - [work items](https://github.com/elyra-ai/elyra/milestone/46)   
    - Is the first shipping vehicle for [JupyterLab 4.0](https://github.com/jupyterlab/jupyterlab/projects/12)
    -  https://github.com/elyra-ai/elyra/issues/3017 ([EOL: June 27](https://endoflife.date/python))

## Q 2 2023
- Code cleanup and pipeline schema updates
  - [work items](https://github.com/elyra-ai/elyra/milestone/70)
  - Includes breaking changes, and function deprecations

## Q3 2023
- Enable scheduling of pipeline runs
  - [work items](https://github.com/elyra-ai/elyra/milestone/71)

## Q4 2023
- Promote `local` mode to full fledged pipeline runtime
  - [work items](https://github.com/elyra-ai/elyra/milestone/72)
  - Expose "local" as a runtime without expanding the existing functionality (https://github.com/elyra-ai/elyra/issues/3036)

**Comment by kevin-bates** [comment]
> Remove support for Python 3.7

The Jupyter stack is removing support for python 3.7 in their upcoming (eminent) major releases on which Elyra 4 will depend (e.g., jupyter_server 2 and jupyter_client 8).  As a result, I think we can move this item into Q1.

**Reply by ptitzler** [reply]
Makes sense. I changed it and will open an issue.
